frontend/src/concepts/pipelines/content/pipelineSelector/utils.ts

```diff
@@ -0,0 +1,88 @@
+import { SearchInputProps } from '@patternfly/react-core';
+import * as React from 'react';
+import { GetColumnSort, SortableData } from '~/components/table';
+import usePipelineFilter, {
+  FilterOptions,
+  getDataValue,
+} from '~/concepts/pipelines/content/tables/usePipelineFilter';
+import { PipelinesFilter } from '~/concepts/pipelines/types';
+
+type SelectorTableColumnSortProps<DataType> = {
+  columns: SortableData<DataType>[];
+  sortField?: string;
+  setSortField: (field: string) => void;
+  sortDirection?: 'asc' | 'desc';
+  setSortDirection: (dir: 'asc' | 'desc') => void;
+};
+
+export const getSelectorTableColumnSort =
+  <T>({
+    columns,
+    sortField,
+    sortDirection,
+    setSortField,
+    setSortDirection,
+  }: SelectorTableColumnSortProps<T>): GetColumnSort =>
+  (columnIndex: number) =>
+    columns[columnIndex].sortable
+      ? {
+          sortBy: {
+            index: columns.findIndex((c) => c.field === sortField),
+            direction: sortDirection,
+            defaultDirection: 'asc',
+          },
+          onSort: (_event, index, direction) => {
+            setSortField(String(columns[index].field));
+            setSortDirection(direction);
+          },
+          columnIndex,
+        }
+      : undefined;
```

Comment:
Similar code appears 7 times in the code base. It would be good to remove all instances into a single configurable function.